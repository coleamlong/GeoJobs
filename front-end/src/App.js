import "bootstrap/dist/css/bootstrap.min.css";

import About from "./views/About";

const App = () => {
  return (
    <div className="App">
      <About />
    </div>
  );
};

export default App;
