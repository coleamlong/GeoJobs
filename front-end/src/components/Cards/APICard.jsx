import React from "react";

import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";


const APICard = (props) => {
  const { title, image, description, link } = props.apiInfo;

  return (
    <Card
    style= {{backgroundColor: '#f8edeb'}}>
      <Card.Img className="p-4" variant="top" src={image} />
      <Card.Body>
        <Card.Title>{title}</Card.Title>
        <Card.Text>{description}</Card.Text>
      </Card.Body>
      <Card.Footer
      style= {{backgroundColor: 'whitesmoke'}}>
        <Button href={link} class="btn btn-primary stretched-link"
            style= {{backgroundColor: '#3d405b'}}>
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default APICard;
