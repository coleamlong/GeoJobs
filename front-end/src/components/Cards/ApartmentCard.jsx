import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import apartment_placeholder from "../../assets/placeholder/apartment.png";
import {Highlight} from "react-highlight-regex";

const ApartmentCard = (props) => {
  const {
    address,
    bedrooms,
    bathrooms,
    build_year,
    id,
    price,
    property_type,
    sqft,
    images,
  } = props.apartment;

  function highlightText (input) {
    if (props.regex != null) {
      return <Highlight match={props.regex} text={input} />
    }
    return input
  }

  return (
    <Card
    style= {{backgroundColor: '#ffffff'}}>
      <Card.Img
        style={{
          height: "45%",
          width: "100%",
          objectFit: "cover",
        }}
        src = {images !== null ? (images[0]?.img_url ?? apartment_placeholder) : apartment_placeholder}
        // src={images !== null ? images[0]?.img_url : apartment_placeholder}
      ></Card.Img>
      <Card.Text className="text-center">{highlightText(property_type)}</Card.Text>
      <Card.Title className="text-center text-uppercase ">{highlightText(address)}</Card.Title>
      <Card.Body>
        <Card.Subtitle className="text-center">
          {highlightText(String(`${
              bedrooms !== null ? bedrooms : 1
            } bedroom(s), 
            ${
              bathrooms !== null ? bathrooms : 1
            } bathroom(s)`))}
        </Card.Subtitle>
        <Card.Text className="text-center">
          {sqft !== null ? highlightText(`${sqft} sqft -- `) : ""}
          {price !== null ? highlightText(`$${price} / month`) : ""}
        </Card.Text>
        {build_year !== null && (
          <Card.Text className="text-center align-text-bottom">
            Build Year: {highlightText(String(build_year))}
          </Card.Text>
        )}
      </Card.Body>
      <Card.Footer
        style= {{backgroundColor: 'whitesmoke'}}>
        <Button
          className="btn btn-primary stretched-link"
          variant="dark"
          href={`/apartment/${id}`}
          style= {{marginLeft:10, backgroundColor: 'midnightblue'}}
        >
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default ApartmentCard;
