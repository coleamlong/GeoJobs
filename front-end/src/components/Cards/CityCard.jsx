import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { Link } from "react-router-dom";
import { Stack } from "@mui/system";

const CityCard = (props) => {
  return (
    <Stack>
    <Card>
      <Card.Body>
        <Card.Title>{props.name}</Card.Title>
        <Card.Text>State: {props.state}</Card.Text>
        <Card.Text>Population: {props.population}</Card.Text>
        <Card.Text>Rating: {props.rating}</Card.Text>
        <Card.Text>Budget: {props.Bodybudget}</Card.Text>
        <Button Link={"/counties/id=" + props.key}>more info on {props.name}</Button>
      </Card.Body>
    </Card>
    </Stack>
  );
};

export default CityCard;
