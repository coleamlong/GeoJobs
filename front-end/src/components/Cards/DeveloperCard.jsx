import React from 'react'

import Card from 'react-bootstrap/Card'

const DeveloperCard = (props) => {
  const { name, image, gitlab_username, role, bio, commits, issues, unit_tests } = props.devInfo
  return (
    <Card>
      <Card.Img variant='top' src={ image } />
      <Card.Body>
        <Card.Title>{ name }</Card.Title>

        <Card.Subtitle>@{ gitlab_username }</Card.Subtitle>
        <Card.Text>Role: { role }</Card.Text>
        <Card.Text>{ bio }</Card.Text>
      </Card.Body>
      <Card.Footer className='text-muted'>
        Commits:    { commits }     <br />
        Issues:     { issues }      <br />
        Unit Tests: { unit_tests }
      </Card.Footer>
    </Card>
  )
}

export default DeveloperCard