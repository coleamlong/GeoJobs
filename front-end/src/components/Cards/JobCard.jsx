import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";

const JobCard = (props) => {
  const { title, city, state, salary, contractType, key } = props.job;
  return (
    <Card style={{ height: "25rem" }}>
      <Card.Body>
        <Card.Title>Title: {title}</Card.Title>
        <Card.Text>City: {city}</Card.Text>
        <Card.Text>State: {state}</Card.Text>
        <Card.Text>Salary: {salary}</Card.Text>
        <Card.Text>ContractType: {contractType}</Card.Text>
      </Card.Body>
      <Card.Footer className="d-flex justify-content-center">
        <Button variant="dark" href={`/job/${key}`}>
          more info on {props.title}
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default JobCard;
