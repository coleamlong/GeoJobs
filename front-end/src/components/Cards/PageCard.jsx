import React from "react";
import Button from "react-bootstrap/Button";

import Card from "react-bootstrap/Card";

const PageCard = (props) => {
  const {
    pageName,
    pageImage,
    pageDescription,
    pageLink
  } = props.pageInfo;
  return (
    <Card
      style= {{marginRight:30, backgroundColor: 'lightgrey'}} >
      <Card.Img variant="top" src={pageImage} />
      <Card.Body>
        <Card.Title>{pageName}</Card.Title>
        <Card.Text>{pageDescription}</Card.Text>
        <Button
          className="btn btn-primary stretched-link"
          style= {{marginRight:30, backgroundColor: 'burlywood'}}
          href={`${pageLink}`}
        >
          <span style={{fontSize: '16px', color: 'black', font: 'Courier-Oblique' }}>More Info</span>
        </Button>
      </Card.Body>
    </Card>
  );
};

export default PageCard;
