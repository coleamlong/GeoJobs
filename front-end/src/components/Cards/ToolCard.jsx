import React from "react";

import Card from "react-bootstrap/Card";

const ToolCard = (props) => {
  const { title, image, description, link } = props.toolInfo;
  return (
    <Card>
      <Card.Img className="p-4" variant="top" src={image} />
      <Card.Body>
        <Card.Title>{title}</Card.Title>

        <Card.Text>{description}</Card.Text>
        <Card.Link href={link}>{link}</Card.Link>
      </Card.Body>
    </Card>
  );
};

export default ToolCard;
