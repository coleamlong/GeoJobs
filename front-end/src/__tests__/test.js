import renderer from "react-test-renderer";

import JobCard from "../components/Cards/JobCard";
import ApartmentCard from "../components/Cards/ApartmentCard";
import CityCard from "../components/Cards/CityCard";
import APICard from "../components/Cards/APICard";
import DeveloperCard from "../components/Cards/DeveloperCard";
import PageCard from "../components/Cards/PageCard";
import ToolCard from "../components/Cards/ToolCard";

import GlobalNavbar from "../components/GlobalNavbar/GlobalNavbar";

import placeholder from "../assets/placeholder/avatar.png";
import App from "../App";
import RouteSwitch from "../RouteSwitch";

it("JobCard Initial correctly", () => {
  const job = {
    title: "Title",
    city: "City",
    state: "State",
    salary: 100,
    ContractType: "Full Time",
    key: 1,
  };
  const component = renderer.create(<JobCard job={job} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("ApartmentCard initializes correctly", () => {
  const apartment = {
    name: "Name",
    city: "City",
    price: 1000,
    type: "Apartment",
    buildYear: 1998,
    key: 1,
  };
  const component = renderer.create(<ApartmentCard apartment={apartment} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("CityCard initializes correctly", () => {
  const city = {
    name: "Name",
    state: "State",
    Population: 123356,
    rating: 100,
    budget: 1000,
    key: 1,
  };
  const component = renderer.create(<CityCard city={city} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("APICard initializes correctly", () => {
  const apiInfo = {
    title: "Title",
    image: placeholder,
    Description: "Description",
    link: "https://www.google.com",
  };
  const component = renderer.create(<APICard apiInfo={apiInfo} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("ToolCard initializes correctly", () => {
  const toolInfo = {
    title: "Title",
    image: placeholder,
    Description: "Description",
    link: "https://www.google.com",
  };
  const component = renderer.create(<ToolCard toolInfo={toolInfo} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("DeveloperCard initializes correctly", () => {
  const devInfo = {
    name: "Name",
    image: placeholder,
    gitlab_username: "GitLab User",
    role: "Role",
    bio: "Bio",
    commits: 0,
    issues: 0,
    unit_tests: 0,
  };
  const component = renderer.create(<DeveloperCard devInfo={devInfo} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("PageCard initializes correctly", () => {
  const pageInfo = {
    pageName: "Name",
    pageImage: placeholder,
    pageDescription: "Description",
    pageLink: "https://www.google.com",
  };
  const component = renderer.create(<PageCard pageInfo={pageInfo} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Navbar initializes correctly", () => {
  const component = renderer.create(<GlobalNavbar />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("App initializes correctly", () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("RouteSwitch initializes correctly", () => {
  const component = renderer.create(<RouteSwitch />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
