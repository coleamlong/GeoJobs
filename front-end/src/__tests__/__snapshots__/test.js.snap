// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APICard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top p-4"
    src="avatar.png"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Title
    </div>
    <p
      className="card-text"
    />
    <a
      className="card-link"
      href="https://www.google.com"
    >
      https://www.google.com
    </a>
  </div>
</div>
`;

exports[`ApartmentCard initializes correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Apartment: 
      Name
    </div>
    <p
      className="card-text"
    >
      Location: 
      City
    </p>
    <p
      className="card-text"
    >
      Price: 
      1000
    </p>
    <p
      className="card-text"
    >
      Type: 
      Apartment
    </p>
    <p
      className="card-text"
    >
      BuldYear: 
      1998
    </p>
    <a
      className="btn btn-dark"
      disabled={false}
      href="/apartment/1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      more info on 
      Name
    </a>
  </div>
</div>
`;

exports[`App initializes correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        GeoJobs
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/about"
            disabled={false}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/jobs"
            disabled={false}
            href="/jobs"
            onClick={[Function]}
          >
            Jobs
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/cities"
            disabled={false}
            href="/cities"
            onClick={[Function]}
          >
            Cities
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/apartments"
            disabled={false}
            href="/apartments"
            onClick={[Function]}
          >
            Apartments
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="main-home"
  >
    <div
      className="bg d-flex justify-content-center"
      style={
        Object {
          "backgroundImage": "url(splash-background.png)",
          "backgroundSize": "cover",
          "height": "100vh",
          "width": "100vw",
        }
      }
    >
      <h1
        className="bg-text"
        style={
          Object {
            "alignItems": "center",
            "backgroundPositionY": "1500",
            "color": "burlywood",
            "justifyContent": "center",
            "marginTop": "300px",
          }
        }
      >
        An easy place to find your dream job, city, and home.
      </h1>
    </div>
    <div
      className="p-4 container"
    >
      <h1
        className="d-flex justify-content-center p-4"
      >
        Where to next?
      </h1>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="job.jpeg"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Jobs
              </div>
              <p
                className="card-text"
              >
                Find your dream job!
              </p>
              <a
                class="btn btn-primary stretched-link"
                href="/jobs"
              >
                Link
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="city.jpeg"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Cities
              </div>
              <p
                className="card-text"
              >
                Find your dream city!
              </p>
              <a
                class="btn btn-primary stretched-link"
                href="/cities"
              >
                Link
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="home.jpeg"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Apartments
              </div>
              <p
                className="card-text"
              >
                Find your dream apartment!
              </p>
              <a
                class="btn btn-primary stretched-link"
                href="/apartments"
              >
                Link
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CityCard initializes correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Name
    </div>
    <p
      className="card-text"
    >
      State: 
      State
    </p>
    <p
      className="card-text"
    >
      Population: 
    </p>
    <p
      className="card-text"
    >
      Rating: 
      100
    </p>
    <p
      className="card-text"
    >
      Budget: 
    </p>
    <a
      className="btn btn-dark"
      disabled={false}
      href="/city/1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      more info on 
      Name
    </a>
  </div>
</div>
`;

exports[`DeveloperCard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="avatar.png"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Name
    </div>
    <div
      className="card-subtitle h6"
    >
      @
      GitLab User
    </div>
    <p
      className="card-text"
    >
      Role: 
      Role
    </p>
    <p
      className="card-text"
    >
      Bio
    </p>
  </div>
  <div
    className="text-muted card-footer"
  >
    Commits: 
    0
     
    <br />
    Issues: 
    0
     
    <br />
    Unit Tests: 
    0
  </div>
</div>
`;

exports[`JobCard Initial correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "height": "25rem",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Title: 
      Title
    </div>
    <p
      className="card-text"
    >
      City: 
      City
    </p>
    <p
      className="card-text"
    >
      State: 
      State
    </p>
    <p
      className="card-text"
    >
      Salary: 
      100
    </p>
    <p
      className="card-text"
    >
      ContractType: 
    </p>
  </div>
  <div
    className="d-flex justify-content-center card-footer"
  >
    <a
      className="btn btn-dark"
      disabled={false}
      href="/job/1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      more info on 
    </a>
  </div>
</div>
`;

exports[`Navbar initializes correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      GeoJobs
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/about"
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/jobs"
          disabled={false}
          href="/jobs"
          onClick={[Function]}
        >
          Jobs
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/cities"
          disabled={false}
          href="/cities"
          onClick={[Function]}
        >
          Cities
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/apartments"
          disabled={false}
          href="/apartments"
          onClick={[Function]}
        >
          Apartments
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`PageCard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="avatar.png"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Name
    </div>
    <p
      className="card-text"
    >
      Description
    </p>
    <a
      class="btn btn-primary stretched-link"
      href="https://www.google.com"
    >
      Link
    </a>
  </div>
</div>
`;

exports[`RouteSwitch initializes correctly 1`] = `
<div
  className="main-home"
>
  <div
    className="bg d-flex justify-content-center"
    style={
      Object {
        "backgroundImage": "url(splash-background.png)",
        "backgroundSize": "cover",
        "height": "100vh",
        "width": "100vw",
      }
    }
  >
    <h1
      className="bg-text"
      style={
        Object {
          "alignItems": "center",
          "backgroundPositionY": "1500",
          "color": "burlywood",
          "justifyContent": "center",
          "marginTop": "300px",
        }
      }
    >
      An easy place to find your dream job, city, and home.
    </h1>
  </div>
  <div
    className="p-4 container"
  >
    <h1
      className="d-flex justify-content-center p-4"
    >
      Where to next?
    </h1>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            className="card-img-top"
            src="job.jpeg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Jobs
            </div>
            <p
              className="card-text"
            >
              Find your dream job!
            </p>
            <a
              class="btn btn-primary stretched-link"
              href="/jobs"
            >
              Link
            </a>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            className="card-img-top"
            src="city.jpeg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Cities
            </div>
            <p
              className="card-text"
            >
              Find your dream city!
            </p>
            <a
              class="btn btn-primary stretched-link"
              href="/cities"
            >
              Link
            </a>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            className="card-img-top"
            src="home.jpeg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Apartments
            </div>
            <p
              className="card-text"
            >
              Find your dream apartment!
            </p>
            <a
              class="btn btn-primary stretched-link"
              href="/apartments"
            >
              Link
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ToolCard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top p-4"
    src="avatar.png"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Title
    </div>
    <p
      className="card-text"
    />
    <a
      className="card-link"
      href="https://www.google.com"
    >
      https://www.google.com
    </a>
  </div>
</div>
`;
